libs/librepcb/projecteditor/boardeditor/boardeditor.cpp

```diff
@@ -861,17 +862,36 @@ QStringList BoardEditor::getSearchToolBarCompleterList() noexcept {
 
 void BoardEditor::goToDevice(const QString& name) noexcept {
   if (Board* board = getActiveBoard()) {
+    if (mLastSearch != name) {
+      mLastSearch = name;
+      mNextDeviceIndex = 0;
+    }
+    QList<BI_Device*> deviceCandidates = {};
     foreach (BI_Device* device, board->getDeviceInstances()) {
-      if (device->getComponentInstance().getName()->toLower() ==
-          name.toLower()) {
-        board->clearSelection();
-        device->setSelected(true);
-        QRectF rect   = device->getFootprint().getBoundingRect();
-        qreal  margin = Length(1000000).toPx();
-        rect.adjust(-margin, -margin, margin, margin);
-        mGraphicsView->zoomToRect(rect);
-        return;
+      if (device->getComponentInstance().getName()
+          ->startsWith(name, Qt::CaseInsensitive)) {
+        deviceCandidates.append(device);
+      }
+    }
+    std::sort(deviceCandidates.begin(), deviceCandidates.end(),
+              [](BI_Device* A, BI_Device* B){
+      return A->getComponentInstance().getName()
+          < B->getComponentInstance().getName();
+    });
```

Comment:
Just to comply with naming conventions:

```suggestion
    std::sort(deviceCandidates.begin(), deviceCandidates.end(),
              [](BI_Device* a, BI_Device* b){
      return a->getComponentInstance().getName()
          < b->getComponentInstance().getName();
    });
```

Reply:
Always the lower case variables... :D